client/src/views/client-area/personal-data/AuthenticationModal.vue

```diff
@@ -3,31 +3,68 @@
 <template>
   <ion-page class="modal">
     <ms-modal
-      :title="`${translationPrefix}.title`"
+      :title="labels.title"
+      :subtitle="labels.subtitle"
       :close-button="{ visible: true }"
       :confirm-button="{
-        label: `${translationPrefix}.nextButton`,
-        disabled: !isFormValid(),
+        label: labels.button,
+        disabled: !canProgress || querying || error.length > 0,
         onClick: submit,
       }"
     >
       <div class="modal-container">
-        <ms-input
-          type="email"
-          v-model="newEmailRef"
-          @change="error = ''"
-          @on-enter-keyup="submit"
-          :label="`${translationPrefix}.newEmail`"
-          :validator="newEmailValidator"
-          ref="newEmailInput"
-        />
-        <ms-password-input
-          v-model="passwordRef"
-          @change="error = ''"
-          @on-enter-keyup="submit"
-          :label="`${translationPrefix}.password`"
-        />
+        <div
+          class="email-container"
+          v-show="step === Steps.NewEmail"
+        >
+          <ms-input
+            type="email"
+            v-model="newEmailRef"
+            @change="error = ''"
+            @on-enter-keyup="submit"
+            :label="`${translationPrefix}.newEmail`"
+            :validator="newEmailValidator"
+            ref="newEmailInput"
+          />
+        </div>
+        <div
+          class="password-container"
+          v-show="step === Steps.Password"
+        >
+          <ms-password-input
+            v-model="passwordRef"
+            @change="error = ''"
+            @on-enter-keyup="submit"
+            :label="`${translationPrefix}.password`"
+            ref="passwordInput"
+          />
+        </div>
+        <div
+          class="code-container"
+          v-show="step === Steps.Code"
+        >
+          <ms-code-validation-input
```

Comment:
This is not focused as the modal opens. I guess the component needs some tweaking in megashark-lib, currently the focus is done 200ms after `mount`.